decentraland/constants/MyAssetStatus.js

```diff
@@ -0,0 +1,7 @@
+export const BUYING = "BUYING";
+export const BOUGHT = "BOUGHT";
+
+export default {
+  BUYING,
+  BOUGHT,
+};
```

Comment:
We'll definitely want a failed one here too

Reply:
I agree, but for now, we aren't tracking error events, should I add to the Constant anyway?

Reply:
Maybe just a TODO in that file rather than actually adding it.

Reply:
Even if it's not an error from the contract we can track, we might use it if the phone is offline and the request itself fails to be made

Reply:
I was thinking about that and these statuses were modeled to be used on the assets on the `MyLandScreen`.
If a purchase failed it'll be added to `LandForSaleScreen` again (it could have status but we aren't using it for now only on `MyLandScreen`). 
Maybe will be better having a status for the action instead of an action for the asset?

Reply:
> Yeah, I think we want a data structure with all relevant actions not related to account funding that closely mirrors that data structure.

So both the my land screen and the list land for sale screen would use the state of that purchase actions data structure to reflect what is shown on its screen.

Reply:
Ok.
We have `assetsForSale` and `myAssets` reducers.
An `assetForSale` stores the `asset` and other sell order info like `price` and `seller`.
On `myAssets` we store only the assets (Since a user could have an asset without buying from a marketplace).
I'm thinking about how will be the best way to store the `status` field. 
We are already storing the `actionId` (txhash) on the `asset` object, I'm thinking to move the status to here and change following your thoughts:
Before: `asset = { id, name, img, actionId, status }`
After: `asset = { id, name, img, action: { id, status } }`
Action status could be: `PENDING`, `FAILED`, `CONFIRMED` or `SUCCESSFULL`.
`CONFIRMED` or `SUCCESSFULL` could be splited soon to distinguish between 1-conf and 7-conf.
And that could be reused for all actions on the app. Make sense?

Reply:
SUCCESSFULL -> SUCCESSFUL

Reply:
> `CONFIRMED` or `SUCCESSFUL` could be splited soon to distinguish between 1-conf and 7-conf.

Love this idea!

Reply:
> After: `asset = { id, name, img, action: { id, status } }`

This looks good to me!

Reply:
> On `myAssets` we store only the assets (Since a user could have an asset without buying from a marketplace).

Makes sense. In this case the actionId would be for the transfer tx instead.

Reply:
Oh wait, there's one thing I'm suggesting that maybe isn't clear.

Let me know if this is too big of a refactor for now.

I think there should be a separate data structure in redux called `userActions`. It's just an object of all the actions the user has taken, keyed maybe by actionId?

(1) myAssets doesn't contain anything about the actions other than maybe actionId as a pointer. Either that OR
(2) `userActions` has a pointer to the assetId for each action and myAssts has nothing about actions.

I think I like option 2 better.

The myAssets screen uses both `userActions` and `myAssets` state from redux to figure out what to show.

The listLandForSale screen uses both `userActions` and `assetsForSale` to figure out what to show.

This is a little more complex at first glance, but this way we're not updating action status for a given assets in two spots, in both `myAssets` and `assetsForSale`

Reply:
Nice!
I like option 2 too because we could have more than one action per asset (e.g.: many attempts before success) tracked too.

Reply:
Yep, good point!

Reply:
This change should generalize well to other apps too, so that’s good

Reply:
> Action status could be: `PENDING`, `FAILED`, `CONFIRMED` or `SUCCESSFUL`

And meta-tx or signed messages for contract-based accounts might have state `BROADCAST` but not yet `PENDING`

Reply:
Done.

Reply:
Awesome, thanks!! Resolving this